### PR TITLE
Loudmouth submission

### PR DESCRIPTION
**Pull request by DrSharky**
Added Loudmouth role.
I simplified some code regarding enqueueing visitors. Specifically the action for enqueue visitors for babushka. The loudmouth works exactly the same way regarding visitor enqueueing action, so I created a separate card that they could both use and created a separate priority for it. Just comment and let me know if you think this is a worse idea.
Also fixed a couple things I saw in the mafia.test.js, as well as added a test for LM.

- [x] Fix game breaking disguiser bug.
- [x] CryOut should get the "absolute" label.
- [x] EnqueueVisitors also needs to be "absolute". I think it is fine for EnqueueVisitors to be absolute because other roles that use card have a second action that can be roleblockable.
- [x] EnqueueVisitors: Don't omit actions with a more negative priority.

### CODE REVIEW

**Comment by ChaoticNeutralGnome** [comment]
The EnqueueVisitors card could also be used for roles like Trickster, Slasher, Santa to check for no visitors, could this PR be prioritized?

**Comment by pikulet** [comment]
this is a fix for #60

**Comment by pikulet** [comment]
breaks when i tried LM guiser game (4 LM 1 guiser, guise and kill your target)

also the CryOut card has                     debugger; on line 17

**Comment by DrSharky** [comment]
> breaks when i tried LM guiser game (4 LM 1 guiser, guise and kill your target) :<
> 
> also the CryOut card has debugger; on line 17

Thanks for testing guiser. I should always try to do that. Also yeah, I forgot about that debugger line, I was testing out something for an IDE.

**Comment by pikulet** [comment]
I didn't look into the reason why it breaks but it might be a guiser bug instead of a LM bug. I just flagged it out specifically because the LM-guiser setup is popular.

**Comment by DrSharky** [comment]
> **LM tests**
> 
> * [x]  visited by no one --> no shouting
> * [x]  visited by watcher --> triggers the shout; although I remember in EM it does not really count, I think roles that are "hidden" shouldn't shout need to check
> * [x]  visited by guiser (yay works!)
> * [x]  killed by 2 mafia --> reveals 1 mafia
> 
> image

Should it reveal 2 mafia if visited by 2?
I'll take a look at hidden role actions, since if we implement Ninja eventually, that shouldn't trigger it. and if it does shout hidden roles right now, then it wouldn't work as intended.

**Comment by ChaoticNeutralGnome** [comment]
It should only reveal 1 maf

**Comment by pikulet** [comment]
You can just add a check for "hidden" when filtering actions. 

And for the mafia kill, I tried using both 2 and 3 mafia. Only 1 will reveal. This is intended because only 1 mafia leads kills on here.

**Comment by pikulet** [comment]
Watcher card currently does not have "hidden" flag anyway. For not revealing ninja, you can add a check filtering away hidden actions.

**Comment by pikulet** [comment]
I realised the Watcher card doesn't have hidden that's why it can be seen. 

Hidden is already filtered by enqueue visitors.

Have to test LM with a hidden action and we should be good to gom

**Comment by DrSharky** [comment]
All tasks are completed. Should be ready for review.

**Comment by pikulet** [comment]
- [x] LM cannot be roleblocked
- [x] LM can detect driver

LGTM

**Review by ChaoticNeutralGnome — APPROVED** [review]

File: Games/types/Mafia/roles/cards/CryOutVisitors.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            {
+                priority: PRIORITY_CRY_VISITORS,
+                labels: ["hidden"],
+                run: function () {
+                    if (this.game.getStateName() != "Night")
+                        return;
+
+                    debugger;
+                    let visitors = this.actor.role.data.visitors;
+                    let names = visitors.map(visitor => visitor.name);
```
[the remark this answers is not in the document]
Comment (DrSharky) [reply Games/types/Mafia/roles/cards/CryOutVisitors.js]: > This is the cause of the guiser bug, for some reason visitors is undefined. An optional chaining operator or pulling names into the if loop will fix it.
> 
> let names = visitors?.map(visitor => visitor.name);
> 
> image

Didn't realize there was still a guiser bug. good on you for finding that.

[the remark this answers is not in the document]
Comment (ChaoticNeutralGnome) [reply Games/types/Mafia/roles/cards/CryOutVisitors.js]: That was weeks ago, you already fixed it

**Review by ChaoticNeutralGnome — APPROVED** [review]

File: Games/types/Mafia/roles/cards/CryOutVisitors.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+                priority: PRIORITY_CRY_VISITORS,
+                labels: ["hidden", "absolute"],
+                run: function () {
+                    if (this.game.getStateName() != "Night")
+                        return;
+
+                    let visitors = this.actor.role.data.visitors;
+                    let names = visitors?.map(visitor => visitor.name);
+
+                    if (visitors) {
```
Comment (ChaoticNeutralGnome) [inline Games/types/Mafia/roles/cards/CryOutVisitors.js:20]: An empty array isn't a falsy value in Javascript. You need to do `if(visitors.length) `

or `if(visitors?.length)` if you want to handle null/undefined

**Review by ChaoticNeutralGnome — APPROVED** [review]

File: Games/types/Mafia/roles/cards/CryOutVisitors.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+const { PRIORITY_CRY_VISITORS } = require("../../const/Priority");
+
+module.exports = class CryOutVisitors extends Card {
+
+    constructor(role) {
+        super(role);
+
+        this.actions = [
+            {
+                priority: PRIORITY_CRY_VISITORS,
```
Comment (pikulet) [inline Games/types/Mafia/roles/cards/CryOutVisitors.js:11]: prefer to just use default here

File: Games/types/Mafia/roles/cards/EnqueueVisitors.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+const Card = require("../../Card");
+const {PRIORITY_VISITORS_ENQUEUE } = require("../../const/Priority");
+
+module.exports = class EnqueueVisitors extends Card{
```
Comment (pikulet) [inline Games/types/Mafia/roles/cards/EnqueueVisitors.js:4]: [just a note] In future I think enqueueVisitors can be added as a function in risk's library

getVisitors()

**Review by pikulet — APPROVED** [review]

**Pull request merged**